webapp/src/containers/Rhs/projectDetails/index.tsx

```diff
@@ -35,14 +33,10 @@ const ProjectDetails = memo((projectDetails: ProjectDetails) => {
     const [showProjectConfirmationModal, setShowProjectConfirmationModal] = useState(false);
     const [showSubscriptionConfirmationModal, setShowSubscriptionConfirmationModal] = useState(false);
     const [subscriptionToBeDeleted, setSubscriptionToBeDeleted] = useState<SubscriptionPayload>();
-    const [showAllSubscriptions, setShowAllSubscriptions] = useState(false);
-    const [subscriptionList, setSubscriptionList] = useState<SubscriptionDetails[]>([]);
     const {currentChannelId} = useSelector((reduxState: GlobalState) => reduxState.entities.channels);
-
-    const project: FetchSubscriptionList = {project: projectName};
-    const {data, isLoading} = getApiState(plugin_constants.pluginApiServiceConfigs.getSubscriptionList.apiServiceName, project);
-    const subscriptionData = data as SubscriptionDetails[];
-
+    const [subscriptionListApiParams, setSubscriptionListApiParams] = useState<FetchSubscriptionList>({
+        project: projectName, channel_id: currentChannelId,
+    });
```

Comment:
If we do something like below, we can avoid writing few lines of code -

```suggestion
const [showAllSubscription, setShowAllSubscriptions] = useState(false);
 const subscriptionListApiParams = useMemo<FetchSubscriptionList>(() => ({
        project: projectName,
        channel_id: showAllSubscriptions ? '' : currentChannelId,
    }), [projectName, currentChannelId, showAllSubscriptions]);
```